src/util/dsp/window.cpp

```diff
@@ -0,0 +1,125 @@
+#include "window.hpp"
+#include <math.h>
+
+namespace otto::util::dsp {
+
+  Window::Window (int size, WindowType type, bool normalize) : windowBuffer(size)
+  {
+      reset (size, type, normalize);
+  }
+
+
+  void Window::reset (int size, WindowType type, bool normalize) noexcept
+  {
+      windowBuffer.resize (size);
+      compute (windowBuffer.begin(), size, type, normalize);
+  }
+
+
+  void Window::compute (double* buffer, int size, WindowType type, bool normalize) noexcept
```

Comment:
Instead of this buffer/size pair, this function should probably take a `gsl::span`. They are a little bit weird, and i might implement my own at some point, but they are definately preferable.